sdks/python/apache_beam/runners/interactive/caching/streaming_cache.py

```diff
@@ -336,7 +344,8 @@ def write(self, values, *labels):
           val = v.SerializeToString()
         else:
           val = v
-        f.write(self._default_pcoder.encode(val) + b'\n')
+          self.save_pcoder(coders.registry.get_coder(type(val)), *labels)
```

Comment:
Why does the coder need to be saved for each value saved to the cache? Shouldn't the coders be known before the pipeline is run?

Reply:
Good point, let me move it to the beginning of `def write`. 

This execution route is actually irrelevant because there doesn't seem to be a use case that directly writes non-TestStreamFileHeader/non-TestStreamFileRecord values with StreamingCache.

Writes with the `sink`(`StreamingCacheSink`) does seem to [use the coder](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/testing/test_stream.py#L673), I'll put a `save_pcoder` there too.

Reply:
Discussed offline, we don't have and will discourage this execution route to write arbitrary typed elements into streaming cache directly. Raise TypeError here.